akka-actor-typed-tests/src/test/java/jdocs/akka/typed/SpawnProtocolDocTest.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (C) 2017-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.typed;
+
+import jdocs.akka.typed.IntroTest.HelloWorld;
+
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+
+//#imports1
+import akka.actor.typed.Behavior;
+import akka.actor.typed.SpawnProtocol;
+import akka.actor.typed.javadsl.Behaviors;
+
+//#imports1
+
+//#imports2
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Props;
+import akka.actor.typed.javadsl.AskPattern;
+import akka.util.Timeout;
+
+//#imports2
+
+public class SpawnProtocolDocTest {
+
+  //#main
+  public abstract static class HelloWorldMain {
+    private HelloWorldMain() {
+    }
+
+    public static final Behavior<SpawnProtocol> main =
+      Behaviors.setup( ctx -> {
+        // Start initial tasks
+        // ctx.spawn(...)
+
+        return SpawnProtocol.behavior();
+      });
```

Comment:
I'd simplify this and just pass the spawn behavior directly to the actor system, I think the two options are mostly mutually exclusive - either you want to create all your application top-actors/subsystems in `setup` or you want to use the spawn behavior (and then two or three snowflakes who want to do both, but maybe not optimize samples for them).

Reply:
I was thinking that our recommendation would still be to put initialization in the guardian behavior like this, but spawning from the outside is still a valid pattern for spawning "per request actors" or similar.

Then there will be those that are just too used with the old system.actorOf that will use it for initialization also.

When I say initialization it can be things like ClusterSharding.spawn that should be done once at startup. Starting cluster singletons is another good example. Why would you start ClusterSharding outside and singletons inside (singletons can't be started from the outside).

Reply:
> Why would you start ClusterSharding outside and singletons inside (singletons can't be started from the outside).

Wrong of me, it's an extension now so that point is not true

Reply:
I think the "normal"/recommended way should be to do it all in `setup` but in some cases you'd want to do it "from the side" (tests, just playing around with it etc.). Could be that I am missing some use case though.

Reply:
and shouldn't we show the recommended way then, since we know such snippets will be copied? most will understand that they can remove the `setup`, but not all will realize that they can add it

Reply:
Yes, I think so. Maybe we need a section before this "Starting up your application" with a guardian-setup starting up different parts, (I think we'd still want the spawning behavior documented though).